application-services/custom/json-logic/main.go

```diff
@@ -88,63 +91,64 @@ func main() {
 	os.Exit(0)
 }
 
-func printXMLToConsole(edgexcontext *appcontext.Context, params ...interface{}) (bool, interface{}) {
-	if len(params) < 1 {
+func printXMLToConsole(ctx interfaces.AppFunctionContext, data interface{}) (bool, interface{}) {
+	if data == nil {
 		// We didn't receive a result
 		return false, nil
 	}
 
-	fmt.Println(params[0].(string))
+	ds, ok := data.(string)
+
+	if !ok {
+		return false, fmt.Errorf("expected a string, got %T", data)
+	}
+
+	fmt.Println(ds)
 
 	// Leverage the built in logging service in EdgeX
-	edgexcontext.LoggingClient.Debug("XML printed to console")
+	ctx.LoggingClient().Debug("XML printed to console")
 
-	edgexcontext.Complete([]byte(params[0].(string)))
+	ctx.SetResponseData([]byte(ds))
 	return false, nil
 }
 
 var precision = 4
 
-func ConvertToReadableFloatValues(edgexcontext *appcontext.Context, params ...interface{}) (bool, interface{}) {
+//ConvertToReadableFloatValues is used to facilitate writing jsonlogic rules
+func ConvertToReadableFloatValues(ctx interfaces.AppFunctionContext, param interface{}) (bool, interface{}) {
 
-	edgexcontext.LoggingClient.Debug("Convert to Readable Float Values")
+	ctx.LoggingClient().Debug("Convert to Readable Float Values")
 
-	if len(params) < 1 {
+	if param == nil {
 		// We didn't receive a result
 		return false, nil
 	}
 
-	event := params[0].(models.Event)
+	event := param.(models.Event)
 	for index := range event.Readings {
-		eventReading := &event.Readings[index]
-		edgexcontext.LoggingClient.Debug(fmt.Sprintf("Event Reading for %s: %s is '%s'", event.Device, eventReading.Name, eventReading.Value))
+		if eventReading, ok := event.Readings[index].(models.SimpleReading); ok {
+			ctx.LoggingClient().Debugf("Event Reading for %s: %s is '%s'", event.DeviceName, eventReading.ResourceName, eventReading.Value)
 
-		data, err := base64.StdEncoding.DecodeString(eventReading.Value)
-		if err != nil {
-			return false, fmt.Errorf("unable to Base 64 decode float32/64 value ('%s'): %s", eventReading.Value, err.Error())
-		}
+			switch eventReading.ResourceName {
+			case "RandomValue_Float32":
+				f, err := strconv.ParseFloat(eventReading.Value, 32)
+				if err != nil {
+					return false, fmt.Errorf("unable to decode float32 value bytes: %s", err.Error())
+				}
 
-		switch eventReading.Name {
-		case "RandomValue_Float32":
-			var value float32
-			err = binary.Read(bytes.NewReader(data), binary.BigEndian, &value)
-			if err != nil {
-				return false, fmt.Errorf("unable to decode float32 value bytes: %s", err.Error())
-			}
+				eventReading.Value = strconv.FormatFloat(f, 'f', precision, 32)
 
-			eventReading.Value = strconv.FormatFloat(float64(value), 'f', precision, 32)
+			case "RandomValue_Float64":
```

Comment:
The resource name from Device Virtual has changed to simply `Float64` 
https://github.com/edgexfoundry/device-virtual-go/blob/main/cmd/res/profiles/device.virtual.float.yaml#L26-L40